internal/template/template_product.go

```diff
@@ -244,7 +244,7 @@ func (p *Product) GenerateSubPathAndValue(fake *faker.Faker, subDocSize int) map
 }
 
 func (p *Product) GetValues(document interface{}) (interface{}, error) {
-	return *document.(*Product), nil
+	return document.(*Product), nil
```

Comment:
GetValues should return a pointer
```suggestion
	return *document.(*Product), nil
```